packages/n/newtondynamics/port/xmake.lua

```diff
@@ -0,0 +1,55 @@
+add_rules("mode.debug", "mode.release")
+
+target("newton")
+    set_kind("$(kind)")
+    set_languages("c89", "cxx11")
+    if is_plat("windows") or is_plat("mingw") then
+        add_defines("_WINDOWS", "_CRT_SECURE_NO_WARNINGS")
+        if is_arch("x86") then
+            add_defines("_WIN_32_VER")
+        else
+            add_defines("_WIN_64_VER")
+        end
+        if is_plat("mingw") then
+            if is_arch("x86") then
+                add_defines("_MINGW_32_VER")
+            else
+                add_defines("_MINGW_64_VER")
+            end
+        end
+    elseif is_plat("linux", "android") then
+        add_syslinks("dl", "pthread")
+        if is_arch("x86") then
+            add_defines("_POSIX_VER")
+        else
+            add_defines("_POSIX_VER_64")
+        end
+        if is_plat("android") then
+            add_defines("_ARM_VER")
+        end
+    elseif is_plat("macosx", "iphoneos") then
+        add_defines("_MACOSX_VER")
+    end
+
+    if is_plat("windows") then
+        if get_config("kind") == "static" then
```

Comment:
is_kind("static")

Reply:
Fixed.
Also, is it normal that this define, while being public, is not included in the package info? I had to use on_load in the package to have it.

Reply:
If you need to export definitions to users, you should set it in the external package scope through add_defines or set it in on_load.

The definition in port/xmake.lua only affects the compilation of the package itself, and does not export the definition to the user project.